fw/cache.c

```diff
@@ -507,7 +512,18 @@ tfw_cache_entry_is_live(TfwHttpReq *req, TfwCacheEntry *ce)
 			lt_min_fresh = ce->lifetime - req->cache_ctl.min_fresh;
 		lt_fresh = min(lt_max_age, lt_min_fresh);
 	}
-	if (!(req->cache_ctl.flags & TFW_HTTP_CC_MAX_STALE)) {
+	/*
+	 * RFC 7234 Section 4.2.4:
+	 * A cache MUST NOT generate a stale response if it is prohibited by an
+	 * explicit in-protocol directive (e.g., by a "no-store" or "no-cache"
+	 * cache directive, a "must-revalidate" cache-response-directive, or an
+	 * applicable "s-maxage" or "proxy-revalidate" cache-response-directive;
+	 * see Section 5.2.2).
+	 */
```

Comment:
@krizhanovsky Can you please explain the difference?
To ignore max-stale client's directive?

Reply:
`s-maxage` can have a value, which would allow to use a stale cached response, but the fact that there is `s-maxage` doesn't allow to use a stale cached response regardless the value of `s-maxage`

Reply:
@krizhanovsky What is "stale" response? A response which age greater than freshness lifetime? Or any response in the cache?

Reply:
@Dmitry-Gouriev stale response is any 'non fresh' response, e.g. which age is greater than the max age. Normally web cache should not service them, but an origin server can be unavailable and there are web cache setting along with the RFC standard, which allows to service stale responses

Reply:
@krizhanovsky in this case how "s-maxage can have a value, which would allow to use a stale cached response"? Response which age is less than s-maxage value is not stale, is it?

Reply:
@Dmitry-Gouriev never mind. Let's just add the comment - there is too much discussion on it.